core/client/src/main/java/alluxio/client/block/AlluxioBlockStore.java

```diff
@@ -43,22 +43,29 @@
   private static AlluxioBlockStore sClient = null;
 
   /**
-   * @return a new instance of Alluxio block store
+   * @return an instance of the Alluxio block store
    */
   public static synchronized AlluxioBlockStore get() {
     if (sClient == null) {
-      sClient = new AlluxioBlockStore();
+      sClient =
+          new AlluxioBlockStore(BlockStoreContext.INSTANCE, NetworkAddressUtils.getLocalHostName());
     }
     return sClient;
   }
 
   private final BlockStoreContext mContext;
+  private final String mLocalHostName;
 
   /**
    * Creates an Alluxio block store.
+   *
+   * @param context the block store context to use for acquiring worker and master clients
+   * @param localHostName the local hostname for the block store
    */
-  private AlluxioBlockStore() {
-    mContext = BlockStoreContext.INSTANCE;
+  public AlluxioBlockStore(BlockStoreContext context, String localHostName) {
```

Comment:
Changed to package-private since it isn't needed elsewhere atm, is that what you meant?